### PR TITLE
MMA8653 Driver BBC Micro:bit

### PR DESCRIPTION
**Pull request by WilliamGFish**
Attempt at creating a MMA8652FC Driver for Micro:bit

### CODE REVIEW

File: drivers/sensor/mma8653fc/mma8653fc.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+	return 0;
+}
+
+static const struct sensor_driver_api mma8653_driver_api = {
+	.sample_fetch = mma8653_sample_fetch,
+	.channel_get = mma8653_channel_get,
+};
+
+int mma8653_init(struct device *dev)
```
Comment (MaureenHelm) [inline drivers/sensor/mma8653fc/mma8653fc.c]: This should be `static`

File: boards/arm/bbc_microbit/bbc_microbit.dts [hunk 2]
```diff
@@ -1,77 +1,89 @@
...
-		};
-
-#if defined(CONFIG_FS_FLASH_STORAGE_PARTITION)
-		storage_partition: partition@3e000 {
-			label = "storage";
-			reg = <0x0003e000 0x00002000>;
-		};
-#endif
-	};
-};
```
Comment (MaureenHelm) [inline boards/arm/bbc_microbit/bbc_microbit.dts]: Why are there so many changes in this file? Did you convert line endings? I'd expect the only change to be adding the sensor node to the i2c bus.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
This is a really good start, thank you for your contribution!

There are few things missing to make this PR mergeable:
1. Add an entry for the new sensor to `drivers/sensor/Kconfig` and `drivers/sensor/CMakeLists.txt`
2. Add this sensor to `tests/drivers/build_all/sensors_i_z.conf` to ensure the driver gets built by CI
3. Fix your commit messages according to the [guidelines](https://docs.zephyrproject.org/latest/contribute/contribute_guidelines.html). This will fix the CI failure.

File: boards/arm/bbc_microbit/bbc_microbit.dts (hunk 2 above)
Comment (WilliamGFish) [reply boards/arm/bbc_microbit/bbc_microbit.dts]: I'm not sure if the i2c config was correct, looking for advice.

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Not necessary, FXOS8700 driver can be used for this device.

**Comment by WilliamGFish** [comment]
Closed as been advised FXOS8700 driver can be used for this device.

**Comment by jfischer-phytec-iot** [comment]
@WilliamGFish please take a loot at #10154

**Pull request closed without merging**